libs/ledger/src/chaincode/smart_contract_manager.cpp

```diff
@@ -120,11 +121,12 @@ Contract::Result SmartContractManager::OnCreate(chain::Transaction const &tx)
   nonce = FromBase64(nonce);
   chain::Address const payable_address{crypto::Hash<crypto::SHA256>(tx.from().address() + nonce)};
```

Comment:
I know that it is not part of this PR but can we rename this to contract address because I think that this will make this code simpler to reason about